### ISSUE TITLE
FPS Tanking In Specific Menu in Fire Emblem: Three Houses

### ISSUE THREAD

**Issue by KakarottoCake** [Ryujinx/Ryujinx#3549]
### What's the issue you encountered?

The FPS completely tanks at this specific screen in Fire Emblem Three Houses. It was really hard to miss when playing using the 60 fps mod, the game just lags only here, nowhere else. 

### How can the issue be reproduced?

Go to the menu displaying the calendar while in the Monastery

### Environment?

- Ryujinx version: 1.1.207</br>
- Game version: 1.2.0</br>

- System Specs:
  - OS: *Windows 11*
  - CPU: *Ryzen 7 5800X*
  - GPU: *Nvidia RTX 2080 Ti*
  - RAM: *32 GB Ram*
- Applied Mods : Tried with both with 60 FPS mod and with no mods

![image](https://user-images.githubusercontent.com/61133727/183050075-55491d88-0b23-48bf-8785-e4a44455929f.png)

**Comment by gdkchan** [comment]
Sounds like #2533 but that's suppossed to be fixed already. @JumpmanSr any ideas of why its still happening?

**Comment by GamerzHell9137** [comment]
> Sounds like #2533 but that's suppossed to be fixed already. @JumpmanSr any ideas of why its still happening?

JD has said that the fix was a band aid and that the issue wasn't really fixed with it.

**Comment by gdkchan** [comment]
The PR https://github.com/Ryujinx/Ryujinx/pull/2761 has before and after screenshots with the game running at 15 fps before and 30 fps after. The author also states that it hits "Hits 30 FPS easily". So it does appear to be fixed there. I want to know if that's either a regression, it had improvement but not enough to make it run well on some hardware, or if this slowdown is caused by a different issue.

**Comment by jduncanator** [comment]
> > Sounds like #2533 but that's suppossed to be fixed already. @JumpmanSr any ideas of why its still happening?
> 
> JD has said that the fix was a band aid and that the issue wasn't really fixed with it.

What I meant was that the original change didn't actually fix the underlying problem, it just fixed the symptom (which was low performance on that menu). The underlying problem still exists.

**Comment by JumpmanSr** [comment]
> The PR #2761 has before and after screenshots with the game running at 15 fps before and 30 fps after. The author also states that it "Hits 30 FPS easily". So it does appear to be fixed there. I want to know if that's either a regression, it had improvement but not enough to make it run well on some hardware, or if this slowdown is caused by a different issue.

It's because the original method was a bit faster but unfortunately due to some edge cases and talking it over with other developers a slightly slower caching method was used instead. (This was to prevent any further issues that may arise in the future especially if LDN or anything involving networking gets merged). The issue itself is in the game menu and the code was an optimization for the emulator but on some hardware it may not hit the full fps still. I imagine on a 60 fps patch/cheat this is more noticeable than 30 fps. 

tl;dr No regression, just using a slower method than originally intended for complete compatibility but may not hit 30 fps as easily on all hardware than originally intended. I will look into some potential optimizations or revisit the fast-path as potentially with new .Net revisions they may have fixed edge cases (maybe, hopefully heh).

**Comment by jduncanator** [comment]
The performance overhead comes from the game dispatching service IPC calls hundreds of times per second. No amount of caching or optimisation on the service side will ever improve that, as there is a fixed cost for dispatching an IPC message into the service, and waiting for a response.

I still believe this is a bug in Ryujinx, triggering this behaviour in the game that wouldn't otherwise happen on console. It makes zero sense for the game to be attempting to determine a local interfaces IP address hundreds of times per frame. The only true way to fix the performance issue here is to prevent the game from making the service call in the first place.

**Comment by KakarottoCake** [comment]
I hate to bring back up an old issue, but was this ever fixed?

**Comment by xCISACx** [comment]
Can confirm this still happens on the latest build.

**Comment by KakarottoCake** [comment]
@xCISACx 
Yup, still an issue on Ryujinx, yet yuzu has zero issues with this

**Comment by MetrosexualGarbodor** [comment]
Please report game-specific issues in the [compatibility entry for the game](https://github.com/Ryujinx/Ryujinx-Games-List/issues/230). Thank you.
